### PR TITLE
Minor refactoring and robust mode check for imagenet_utils.py

### PR DESCRIPTION
**Pull request by jaketae**
Minor refactoring of code for better readability of conditional statements (using `elif` instead of multiple `if`s if the statements check for conditions on the same object). Also added robust `mode` argument check to prevent silent failure.

### CODE REVIEW

File: tensorflow/python/keras/applications/imagenet_utils.py [hunk 1]
```diff
@@ -66,11 +66,11 @@
       {ret}
 
   Raises:
-      ValueError: In case of unknown `data_format` argument.
+      ValueError: In case of unknown `mode` or `data_format` argument.
```
Comment (pavithrasv) [inline tensorflow/python/keras/applications/imagenet_utils.py]: This cannot be added here as this doc is shared across all applications. This will have to be made a placeholder similar to {mode} / {ret} above.

File: tensorflow/python/keras/applications/imagenet_utils.py [hunk 2]
```diff
@@ -97,9 +97,12 @@
 @keras_export('keras.applications.imagenet_utils.preprocess_input')
 def preprocess_input(x, data_format=None, mode='caffe'):
   """Preprocesses a tensor or Numpy array encoding a batch of images."""
+  if mode not in {'caffe', 'tf','torch'}:
+    raise ValueError('Unknown mode ' + str(mode))
+
```
Comment (pavithrasv) [inline tensorflow/python/keras/applications/imagenet_utils.py]: + space

File: tensorflow/python/keras/applications/imagenet_utils.py [hunk 3]
```diff
@@ -414,10 +415,10 @@ def validate_activation(classifier_activation, weights):
     return
 
   classifier_activation = activations.get(classifier_activation)
-  if classifier_activation not in [
+  if classifier_activation not in {
```
Comment (pavithrasv) [inline tensorflow/python/keras/applications/imagenet_utils.py]: Why do you want to change this to a dict?

File: tensorflow/python/keras/applications/imagenet_utils.py [hunk 4]
```diff
@@ -97,9 +97,12 @@
 @keras_export('keras.applications.imagenet_utils.preprocess_input')
 def preprocess_input(x, data_format=None, mode='caffe'):
   """Preprocesses a tensor or Numpy array encoding a batch of images."""
+  if mode not in {'caffe', 'tf','torch'}:
+    raise ValueError('Unknown mode ' + str(mode))
```
Comment (pavithrasv) [inline tensorflow/python/keras/applications/imagenet_utils.py]: Can you add a test for this?

**Review by pavithrasv — COMMENTED** [review]
Thank you for the PR!

File: tensorflow/python/keras/applications/imagenet_utils.py (hunk 3 above)
Comment (jaketae) [reply tensorflow/python/keras/applications/imagenet_utils.py]: This was just a very minor edit made based on the fact that the `in` operator works faster on sets (not dict) than lists. I can write a `timeit` test if you want, but this is already pretty well-documented on this [stackoverflow post](https://stackoverflow.com/questions/2831212/python-sets-vs-lists). Thanks!

Comment (pavithrasv) [reply tensorflow/python/keras/applications/imagenet_utils.py]: Thanks for the note! We don't need tests for this.

File: tensorflow/python/keras/applications/imagenet_utils.py (hunk 4 above)
Comment (jaketae) [reply tensorflow/python/keras/applications/imagenet_utils.py]: Can you give me a little guidance as to what kind of test I should be writing? I'm new to contributing, a little pointer would help me a lot. Thank you!

Comment (pavithrasv) [reply tensorflow/python/keras/applications/imagenet_utils.py]: Please take a look at https://github.com/tensorflow/tensorflow/blob/aa070f5be6bfc55c25802540da149138696f965f/tensorflow/python/keras/applications/imagenet_utils_test.py#L29

We can add a simple test case for invalid mode here.

File: tensorflow/python/keras/applications/imagenet_utils.py (hunk 1 above)
Comment (jaketae) [reply tensorflow/python/keras/applications/imagenet_utils.py]: I was thinking that the `mode`  value error should be in the comments across all variations because it isn't something specific to a certain mode, just like the `data_format` value error. From what I see, the {mode} placeholder exists because a different docstring is needed for each mode. Is this still the case? (Please let me know if I'm missing something, I'm probably not understanding a detail here.)

File: tensorflow/python/keras/applications/imagenet_utils.py (hunk 4 above)
Comment (jaketae) [reply tensorflow/python/keras/applications/imagenet_utils.py]: Before I make a new PR for imagenet_utils_test.py, is this similar to what you had in mind?
```python
with self.assertRaises(ValueError):
    utils.preprocess_input(x, mode='some_unknown_mode')
```

Comment (pavithrasv) [reply tensorflow/python/keras/applications/imagenet_utils.py]: Yes

File: tensorflow/python/keras/applications/imagenet_utils.py (hunk 1 above)
Comment (pavithrasv) [reply tensorflow/python/keras/applications/imagenet_utils.py]: Please checkout how mobilenet reuses this docstring. `mode` is not an input in the `preprocess_input` function of mobilenet.

File: tensorflow/python/keras/applications/imagenet_utils.py (hunk 4 above)
Comment (pavithrasv) [reply tensorflow/python/keras/applications/imagenet_utils.py]: Yes.

File: tensorflow/python/keras/applications/imagenet_utils.py (hunk 1 above)
Comment (jaketae) [reply tensorflow/python/keras/applications/imagenet_utils.py]: Thanks for the pointer. I made changes to the docstring based on your comment, specifically by adding a placeholder and reflecting this change across all models. Please take a look at let me know what you think!

**Review by pavithrasv — DISMISSED** [review]

**Comment by gbaned** [comment]
@jaketae Can you please check build failures. Thanks!

**Comment by jaketae** [comment]
@gbaned Thank you for the friendly reminder. I realized that one of the files had a missing argument required for docstring formatting. I've pushed a new commit, so hopefully it goes through.

**Comment by gbaned** [comment]
@jaketae Could you please resolve the conflicts? Thanks!

File: tensorflow/python/keras/applications/mobilenet_v2.py [hunk 5]
```diff
@@ -500,14 +500,42 @@ def _make_divisible(v, divisor, min_value=None):
...
+
+  Arguments
+    x: A 4D numpy array consists of RGB values within [0, 255].
+
+  Returns
+    Preprocessed array.
+
+  Raises
+    ValueError: In case of unknown `data_format` argument.
+  """
```
Comment (pavithrasv) [inline tensorflow/python/keras/applications/mobilenet_v2.py:513]: Why is this needed? We are overriding the doc below right?

Same for decode_predictions

Comment (jaketae) [reply tensorflow/python/keras/applications/mobilenet_v2.py]: My bad, you're right. I just pushed a commit removing the unnecessary docs. Thank you for the review!

**Review by pavithrasv — APPROVED** [review]
